src/test/tcp/test_tcp.c

```diff
@@ -212,9 +258,18 @@ static int _do_connect(int fd, struct sockaddr_in* serveraddr, iowait_func iowai
     return 0;
 }
 
-static int _do_serve(int fd, struct sockaddr_in* bindaddr, iowait_func iowait, int* clientsdout) {
+static void send_port_in_queue(int fd, struct sockaddr_in* bindaddr, const char *queuename){
+    socklen_t len = sizeof(bindaddr);
```

Comment:
You're getting the size of the pointer, not the size of the struct. This "luckily" happens to work since pointers are 8 bytes long here :).

Recommend changing to `sizeof(*bindaddr)`

Reply:
oh yes of course !